test/functionalities/Functionality.test.js

```diff
@@ -283,7 +284,67 @@ describe('the Functionality class', () => {
         ])
       })
     })
+  })
+
+  describe('when calling .deriveUnit on a slot', () => {
+    describe('on a slot that has NOT "any" as unit', () => {
+      it('returns the unit of the slot', () => {
+        const unit = '°C'
+        const slot = FloatInSlotSeeder.seedOne({ unit })
+        const heater = HeaterActuatorSeeder.seedOne({ slots: [ slot ] })
+
+        const slotWithinFct = heater.getSlotByName(slot.name)
+
+        expect(slotWithinFct.derivedUnit).toBe(unit)
+      })
+    })
+
+    describe('on a slot that has "any" as unit', () => {
+      describe('when the slot is an InSlot', () => {
+        describe('when it has no connection', () => {
+          it('returns "any"', () => {
+            const unit = 'any'
+            const slot = FloatInSlotSeeder.seedOne({ unit })
+            const controller = PIDControllerSeeder.seedOne({ slots: [ slot ] })
+
+            const slotWithinFct = controller.getSlotByName(slot.name)
+
+            expect(slotWithinFct.derivedUnit).toBe(unit)
+          })
+        })
+
+        describe('when it has a connection', () => {
+          it('returns the unit of its source', () => {
```

Comment:
```suggestion
          it('returns the derived unit of its source', () => {
```
(assuming this updated test description is accurate?)

Reply:
fixed